lib/PSR/Events/Interface.hs

```diff
@@ -50,11 +56,40 @@ data EventFilterParams = EventFilterParams
     }
     deriving (Generic)
 
-data Events = Events 
+data Events = Events
     { addExecutionEvent :: BlockHeader -> ExecutionEventPayload -> IO ()
     , addCancellationEvent :: BlockHeader -> ScriptHash -> IO ()
     , addSelectionEvent :: BlockHeader -> IO ()
-    , getEventsChannel :: TChan Event 
+    , getEventsChannel :: TChan Event
     , getEvents :: EventFilterParams -> IO [Event]
     }
 
+eventMatchesFilter :: EventFilterParams -> Event -> Bool
+eventMatchesFilter (EventFilterParams typ time_begin time_end slot_begin slot_end _limit _offset name_or_script_hash) event =
+    and
+        [ check (event.eventType ==) typ
+        , check (event.createdAt >=) time_begin
+        , check (event.createdAt <=) time_end
+        , check (slotNo >=) slot_begin
+        , check (slotNo <=) slot_end
+        , or
+            [ isNothing name_or_script_hash
+            , name_or_script_hash == mScriptName
+            , name_or_script_hash == mScriptHashText
+            ]
```

Comment:
Either the name hasn't been set, so always succeed, or the optional values below must be `Just some_text`.